companion.py

```diff
@@ -355,33 +396,39 @@ def query(self, endpoint):
             self.retrying = False
             self.login()
             raise CredentialsError()
+
         elif 500 <= r.status_code < 600:
             # Server error. Typically 500 "Internal Server Error" if server is down
             self.dump(r)
-            raise ServerError()
+            raise ServerError('Received error {} from server'.format(r.status_code))
 
         try:
-            r.raise_for_status()	# Typically 403 "Forbidden" on token expiry
-            data = r.json()		# May also fail here if token expired since response is empty
-        except:
+            r.raise_for_status()  # Typically 403 "Forbidden" on token expiry
+            data = r.json()  # May also fail here if token expired since response is empty
+
+        except (requests.HTTPError, ValueError) as e:
```

Comment:
Perhaps `json.JSONDecodeError` instead of `ValueError`.  Heck, separate `except ..:` clause so that use can be made of the specific properties if needs be ?

Reply:
Oh, hang on, that's still in *requests* not the actual json decode.

Reply:
Except that *is* calling through to `json.loads()` if my reading of pydoc for `requests.Response.json` is correct.  But it must be catching the json specific exception in order to raise the ValueError.  So yeah, what you have is what it will need to be.

It's that or we explicitly call `json.loads()` ourselves on the r.content.  Leave it as is for now, we can start fiddling if we ever see the need for more fine-detailed error reporting.